Sources/Swift/Helper/SentryExtraContextProvider.swift

```diff
@@ -0,0 +1,73 @@
+@_implementationOnly import _SentryPrivate
+
+@_spi(Private) @objc public final class SentryExtraContextProvider: NSObject {
+    
+    private static let kSentryProcessInfoThermalStateNominal = "nominal"
+    private static let kSentryProcessInfoThermalStateFair = "fair"
+    private static let kSentryProcessInfoThermalStateSerious = "serious"
+    private static let kSentryProcessInfoThermalStateCritical = "critical"
+    
+    private let crashWrapper: SentryCrashWrapper
+    private let processInfoWrapper: SentryProcessInfoSource
+    
+    #if (os(iOS) || targetEnvironment(macCatalyst)) && !SENTRY_NO_UIKIT
+    private let deviceWrapper: SentryUIDeviceWrapper
+
+    @objc public init(crashWrapper: SentryCrashWrapper, processInfoWrapper: SentryProcessInfoSource, deviceWrapper: SentryUIDeviceWrapper) {
+        self.crashWrapper = crashWrapper
+        self.processInfoWrapper = processInfoWrapper
+        self.deviceWrapper = deviceWrapper
+    }
+    #else
+    @objc public init(crashWrapper: SentryCrashWrapper, processInfoWrapper: SentryProcessInfoSource) {
+        self.crashWrapper = crashWrapper
+        self.processInfoWrapper = processInfoWrapper
+    }
+    #endif
+    
+    @objc public func getExtraContext() -> [AnyHashable: Any] {
```

Comment:
`l`: We could even be a bit more specific
```suggestion
    @objc public func getExtraContext() -> [String: Any] {
```